core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,638 @@
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      const typename mdspan_type::mapping_type &arg_mapping) {
+    constexpr bool has_exec = Impl::ViewCtorProp<P...>::has_execution_space;
+    // Copy the input allocation properties with possibly defaulted properties
+    // We need to split it in two to avoid MSVC compiler errors
+    auto prop_copy_tmp =
+        Impl::with_properties_if_unset(arg_prop, std::string{});
+    auto prop_copy = Impl::with_properties_if_unset(
+        prop_copy_tmp, memory_space{}, execution_space{});
+    using alloc_prop = decltype(prop_copy);
+
+    if (alloc_prop::initialize &&
+        !alloc_prop::execution_space::impl_is_initialized()) {
+      // If initializing view data then
+      // the execution space must be initialized.
+      Kokkos::Impl::throw_runtime_exception(
+          "Constructing View and initializing data with uninitialized "
+          "execution space");
+    }
+    return data_handle_type(Impl::make_shared_allocation_record<ElementType>(
+        arg_mapping.required_span_size(),
+        Impl::get_property<Impl::LabelTag>(prop_copy),
+        Impl::get_property<Impl::MemorySpaceTag>(prop_copy),
+        has_exec ? std::optional{Impl::get_property<Impl::ExecutionSpaceTag>(
+                       prop_copy)}
+                 : std::optional<execution_space>{std::nullopt},
+        std::integral_constant<bool, alloc_prop::initialize>(),
+        std::integral_constant<bool, alloc_prop::sequential_host_init>()));
+  }
+
+ public:
+  template <class... P>
+  // requires(!Impl::ViewCtorProp<P...>::has_pointer)
+  explicit inline BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
+      : BasicView(create_data_handle(arg_prop, arg_mapping), arg_mapping) {}
+
+  template <class... P>
+  // requires(Impl::ViewCtorProp<P...>::has_pointer)
+  KOKKOS_FUNCTION explicit inline BasicView(
+      const Impl::ViewCtorProp<P...> &arg_prop,
+      std::enable_if_t<Impl::ViewCtorProp<P...>::has_pointer,
+                       typename mdspan_type::mapping_type> const &arg_mapping)
+      : BasicView(
+            data_handle_type(Impl::get_property<Impl::PointerTag>(arg_prop)),
+            arg_mapping) {}
+
+ protected:
+  template <class OtherElementType, class OtherExtents, class OtherLayoutPolicy,
+            class OtherAccessorPolicy, class... SliceSpecifiers>
+  KOKKOS_INLINE_FUNCTION BasicView(
+      Impl::subview_ctor_tag_t,
+      const BasicView<OtherElementType, OtherExtents, OtherLayoutPolicy,
+                      OtherAccessorPolicy> &src_view,
+      SliceSpecifiers... slices)
+      : BasicView(submdspan(
+            src_view.to_mdspan(),
+            Impl::transform_kokkos_slice_to_mdspan_slice(slices)...)) {}
+
+ public:
+  //----------------------------------------
+  // Conversion to MDSpan
```

Comment:
no, I have not written maximum number of tests on these implementation details. Most of this comes indirectly through View in the end. We never had tests for all implementation details, and I am not sure it's worthwhile to actually implement all View tests for BasicView too, just because we use it as a separate base class.